Albums functionality

surinder-tsys opened the pull request:
This PR contains the new Albums Functionality.

Library PR: https://github.com/nextcloud/android-library/pull/1736

<!--
TESTING

Writing tests is very important. Please try to write some tests for your PR. 
If you need help, please do not hesitate to ask in this PR for help.

Unit tests: https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#unit-tests
Instrumented tests: https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#instrumented-tests
UI tests: https://github.com/nextcloud/android/blob/master/CONTRIBUTING.md#ui-tests
 -->
- [x] Tests written, or not not needed

surinder-tsys committed:
Albums functionality.

Signed-off-by: A117870935 <surinder.kumar@t-systems.com>

surinder-tsys committed:
Albums functionality.

Signed-off-by: A117870935 <surinder.kumar@t-systems.com>

# Conflicts:
#	app/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java
#	app/src/main/java/com/owncloud/android/ui/adapter/GalleryAdapter.kt

# Conflicts:
#	app/src/main/res/values/strings.xml

# Conflicts:
#	app/src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java

# Conflicts:
#	app/src/main/java/com/nextcloud/client/di/ComponentsModule.java

# Conflicts:
#	app/src/main/java/com/owncloud/android/ui/adapter/OCFileListDelegate.kt

surinder-tsys committed:
Albums functionality.

Signed-off-by: A117870935 <surinder.kumar@t-systems.com>

# Conflicts:
#	app/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java
#	app/src/main/java/com/owncloud/android/ui/adapter/GalleryAdapter.kt

# Conflicts:
#	app/src/main/res/values/strings.xml

# Conflicts:
#	app/src/main/java/com/owncloud/android/ui/fragment/OCFileListFragment.java

# Conflicts:
#	app/src/main/java/com/nextcloud/client/di/ComponentsModule.java

# Conflicts:
#	app/src/main/java/com/nextcloud/ui/albumItemActions/AlbumItemAction.kt
#	app/src/main/java/com/nextcloud/ui/albumItemActions/AlbumItemActionsBottomSheet.kt
#	app/src/main/java/com/owncloud/android/ui/activity/AlbumsPickerActivity.kt
#	app/src/main/java/com/owncloud/android/ui/adapter/albums/AlbumFragmentInterface.kt
#	app/src/main/java/com/owncloud/android/ui/adapter/albums/AlbumGridItemViewHolder.kt
#	app/src/main/java/com/owncloud/android/ui/adapter/albums/AlbumItemViewHolder.kt
#	app/src/main/java/com/owncloud/android/ui/adapter/albums/AlbumListItemViewHolder.kt
#	app/src/main/java/com/owncloud/android/ui/adapter/albums/AlbumsAdapter.kt
#	app/src/main/java/com/owncloud/android/ui/dialog/CreateAlbumDialogFragment.kt
#	app/src/main/java/com/owncloud/android/ui/fragment/GalleryFragment.java
#	app/src/main/java/com/owncloud/android/ui/fragment/albums/AlbumItemsFragment.kt
#	app/src/main/java/com/owncloud/android/ui/fragment/albums/AlbumsFragment.kt
#	app/src/main/res/layout/albums_fragment.xml
#	app/src/main/res/layout/albums_grid_item.xml
#	app/src/main/res/layout/albums_list_item.xml
#	app/src/main/res/menu/bottom_navigation_menu.xml
#	app/src/main/res/menu/partial_drawer_entries.xml

surinder-tsys committed:
Rename .java to .kt

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
Rename .java to .kt

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
Code refactoring and operation migrated to kotlin

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
Code refactoring and operation migrated to kotlin

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
Fix -> video click event, overflow menu item hide and theming add menu

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
Fix: preview album items

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
upload files to album, hide video overlay icon, minor ui and bug fixes

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
upload files to album, hide video overlay icon, minor ui and bug fixes

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
upload files to album, hide video overlay icon, minor ui and bug fixes

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

tobiasKaminsky committed:
wip

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

# Conflicts:
#	gradle/libs.versions.toml

tobiasKaminsky committed:
small fix

Signed-off-by: tobiasKaminsky <tobias@kaminsky.me>

alperozturk96 reviewed (dismissed):
Hello

Thank you for the PR.

surinder-tsys committed:
code fixes

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

surinder-tsys committed:
album upload worker state handled via localbroadcastmanager

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

alperozturk96 committed:
simplify upload file calls

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
separate worker parameters

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
separate params

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
simplify album fragment

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
use optional generic function to access fragment

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
fix code analytics, simplify

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 commented:
@kra-mo Could you please check this screen?

1. Album overview screen looks like to me a bit plain and empty. Paddings, title font and it looks like Android 6 era. 
2. Media and Album bottom bar navigation view icons are same I think we should distinquish them.

<img width="720" height="1280" alt="Screenshot_20260203_102344" src="https://github.com/user-attachments/assets/6c0f9f40-b0d9-43c2-893c-007656d9ac30" />

alperozturk96 committed:
m3 card

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
m3 card

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
use fab

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
use action bottom sheet for album item actions

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
use action bottom sheet for album item actions

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
fix fab button appereance

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

kra-mo commented on app/src/main/res/values/strings.xml:
```diff
@@ -1505,4 +1506,30 @@
+    <string name="album_already_exists">Album already exists</string>
+    <string name="album_picker_toolbar_title">Pick album</string>
+    <string name="media_picker_toolbar_title">Pick media files</string>
+    <string name="empty_albums_title">Create Albums for your Photos</string>
```
```suggestion
    <string name="empty_albums_title">Create albums for your photos</string>
```

This should also be sentence-case. Material design does not use title case.

alperozturk96 committed:
fix git conflict

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
small typo fixes

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
use outlined card

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
album items fragment code cleanup

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
fix(spotbugs): NAB: Needless Autoboxing

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
fix menu item highlighting

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
adopt overlay manager

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
add albumOperationListener

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

surinder-tsys committed:
rebased with master

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>

alperozturk96 committed:
wip

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
wip

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
wip

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

surinder-tsys committed:
feat(albums): add link sharing and add-to-album

Add a sharing bottom sheet for albums with public link creation, sharing
and revocation, backed by PublicShareLinkAlbumRemoteOperation.

Add files to an album from the file list, the image preview and the media
preview. Album picking moves into FileOperationsHelper.addFileToAlbum, so
AlbumsPickerActivity now receives the selected paths and performs the copy
itself instead of returning them through an activity result.

Signed-off-by: Surinder Kumar <surinder.kumar@t-systems.com>
Assisted-by: ClaudeCode:claude-opus-5

alperozturk96 committed:
wip

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
wip

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>

alperozturk96 committed:
refactor album sharing bottom sheet

Signed-off-by: alperozturk96 <alper_ozturk@proton.me>
